### DISCUSSION TITLE
Mute / Disable all Midi Out and IN

### DISCUSSION THREAD

**Discussion by ransa808** [tttapa/Control-Surface#542] [Q&A]
Is there a way to achieve muting all MIDI CC & Note output? 
And Disabling MIDI IN ?

Thx

**Comment by tttapa** [comment]
This is hard to answer without knowing what you actually want to do.

The simple answer is: use `Control_Surface.disconnectMIDI_Interfaces()`. This stops all MIDI communication. You can later restore it using `Control_Surface.connectDefaultMIDI_Interface()`. 

If you want to “mute” specific types of messages, you can write a custom MIDI Pipe.
